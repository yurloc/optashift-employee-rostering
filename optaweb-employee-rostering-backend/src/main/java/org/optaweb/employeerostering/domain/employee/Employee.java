/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.optaweb.employeerostering.domain.employee;

import java.util.Collection;
import java.util.Set;

import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.persistence.UniqueConstraint;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Pattern;
import javax.validation.constraints.Size;

import org.optaweb.employeerostering.domain.common.AbstractPersistable;
import org.optaweb.employeerostering.domain.contract.Contract;
import org.optaweb.employeerostering.domain.skill.Skill;

@Entity
@Table(uniqueConstraints = @UniqueConstraint(columnNames = {"tenantId", "name"}))
public class Employee extends AbstractPersistable {

    @NotNull
    @Size(min = 1, max = 120)
    @Pattern(regexp = "^(?!\\s).*(?<!\\s)$", message = "Name should not contain any leading or trailing whitespaces")
    private String name;

    @NotNull
    @ManyToOne
    private Contract contract;

    @NotNull
    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(name = "EmployeeSkillProficiencySet",
            joinColumns = @JoinColumn(name = "employeeId", referencedColumnName = "id"),
            inverseJoinColumns = @JoinColumn(name = "skillId", referencedColumnName = "id")
    )
    private Set<Skill> skillProficiencySet;

    @SuppressWarnings("unused")
    public Employee() {
    }

    public Employee(Integer tenantId, String name, Contract contract, Set<Skill> skillProficiencySet) {
        super(tenantId);
        this.name = name;
        this.contract = contract;
        this.skillProficiencySet = skillProficiencySet;
    }

    public boolean hasSkill(Skill skill) {
        return skillProficiencySet.contains(skill);
    }

    public boolean hasSkills(Collection<Skill> skills) {
        return skillProficiencySet.containsAll(skills);
    }

    @Override
    public String toString() {
        return name;
    }

    // ************************************************************************
    // Simple getters and setters
    // ************************************************************************

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Set<Skill> getSkillProficiencySet() {
        return skillProficiencySet;
    }

    public void setSkillProficiencySet(Set<Skill> skillProficiencySet) {
        this.skillProficiencySet = skillProficiencySet;
    }

    public Contract getContract() {
        return contract;
    }

    public void setContract(Contract contract) {
        this.contract = contract;
    }

    @Override
    public boolean equals(Object o) {
        if (o instanceof Employee) {
            Employee other = (Employee) o;
            return this.name.equals(other.getName()) &&
                    this.contract.equals(other.getContract()) &&
                    this.skillProficiencySet.equals(other.getSkillProficiencySet());
        } else {
            return false;
        }
    }

    @Override
    public int hashCode() {
        return 31 * ((31 * name.hashCode()) ^ contract.hashCode()) ^ skillProficiencySet.hashCode();
    }
}
