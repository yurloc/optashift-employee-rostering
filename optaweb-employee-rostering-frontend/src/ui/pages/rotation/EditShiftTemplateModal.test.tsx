/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import { Spot } from 'domain/Spot';
import { Employee } from 'domain/Employee';
import { ShiftTemplate } from 'domain/ShiftTemplate';
import moment from 'moment';
import { useTranslation, WithTranslation } from 'react-i18next';
import {
  EditShiftTemplateModal, Props, shiftTemplateDataToShiftTemplate,
  shiftTemplateToShiftTemplateData,
} from './EditShiftTemplateModal';

describe('Edit Shift Template Modal', () => {
  it('should render correctly when closed', () => {
    const editShiftTemplateModal = shallow(<EditShiftTemplateModal
      {...baseProps}
      isOpen={false}
      onSave={jest.fn()}
      onClose={jest.fn()}
      onDelete={jest.fn()}
    />);
    expect(toJson(editShiftTemplateModal)).toMatchSnapshot();
  });

  it('should render correctly when opened', () => {
    const editShiftTemplateModal = shallow(<EditShiftTemplateModal
      {...baseProps}
      isOpen
      onSave={jest.fn()}
      onClose={jest.fn()}
      onDelete={jest.fn()}
    />);
    expect(toJson(editShiftTemplateModal)).toMatchSnapshot();
  });

  it('should render correctly with a shift template when opened', () => {
    const editShiftTemplateModal = shallow(<EditShiftTemplateModal
      {...baseProps}
      shiftTemplate={baseShiftTemplate}
      isOpen
      onSave={jest.fn()}
      onClose={jest.fn()}
      onDelete={jest.fn()}
    />);
    expect(toJson(editShiftTemplateModal)).toMatchSnapshot();
  });

  it('should update state to match shift template when props change', () => {
    const editShiftTemplateModal = shallow(<EditShiftTemplateModal
      {...baseProps}
      shiftTemplate={baseShiftTemplate}
      isOpen
      onSave={jest.fn()}
      onClose={jest.fn()}
      onDelete={jest.fn()}
    />);

    editShiftTemplateModal.setProps({
      shiftTemplate: baseShiftTemplate,
    });
    expect(editShiftTemplateModal.state('editedValue')).toEqual(
      shiftTemplateToShiftTemplateData(baseShiftTemplate, rotationLength),
    );

    editShiftTemplateModal.setState({
      editedValue: shiftTemplateToShiftTemplateData({
        ...baseShiftTemplate,
        shiftTemplateDuration: moment.duration(1, 'h'),
      }, rotationLength),
    });

    expect(editShiftTemplateModal.state('editedValue')).toEqual(
      shiftTemplateToShiftTemplateData({
        ...baseShiftTemplate,
        shiftTemplateDuration: moment.duration(1, 'h'),
      }, rotationLength),
    );

    editShiftTemplateModal.setProps({
      shiftTemplate: undefined,
    });
    expect(editShiftTemplateModal.state('editedValue')).toEqual({
      tenantId: baseProps.tenantId,
    });
  });

  it('should call onSave iff all required properties are defined', () => {
    const onSave = jest.fn();
    const editShiftTemplateModal = new EditShiftTemplateModal({
      ...baseProps,
      onSave,
      onClose: jest.fn(),
      onDelete: jest.fn(),
      isOpen: false,
    });

    editShiftTemplateModal.state = {
      resetCount: 0,
      editedValue: {

      },
    };
    editShiftTemplateModal.onSave();
    expect(onSave).not.toBeCalled();

    editShiftTemplateModal.state = {
      resetCount: 1,
      editedValue: {
        rotationEmployee: null,
        spot,
      },
    };
    editShiftTemplateModal.onSave();
    expect(onSave).not.toBeCalled();

    editShiftTemplateModal.state = {
      resetCount: 2,
      editedValue: shiftTemplateToShiftTemplateData(baseShiftTemplate, rotationLength),
    };
    editShiftTemplateModal.onSave();
    expect(onSave).toBeCalled();
    expect(onSave).toBeCalledWith(baseShiftTemplate);
  });

  it('should call onSave when the save button is clicked with a completed shift', () => {
    const onSave = jest.fn();
    const editShiftTemplateModal = mount(<EditShiftTemplateModal
      {...baseProps}
      shiftTemplate={baseShiftTemplate}
      isOpen
      onSave={onSave}
      onClose={jest.fn()}
      onDelete={jest.fn()}
    />);

    editShiftTemplateModal.find('button[aria-label="Save"]').simulate('click');
    expect(onSave).toBeCalled();
    expect(onSave).toBeCalledWith(baseShiftTemplate);
  });

  it('should call onClose when the bottom close button is clicked', () => {
    const onClose = jest.fn();
    const editShiftTemplateModal = mount(<EditShiftTemplateModal
      {...baseProps}
      shiftTemplate={baseShiftTemplate}
      isOpen
      onSave={jest.fn()}
      onClose={onClose}
      onDelete={jest.fn()}
    />);

    editShiftTemplateModal.find('button[aria-label="Close Modal"]').simulate('click');
    expect(onClose).toBeCalled();
  });

  it('should call onClose when the top right X close button is clicked', () => {
    const onClose = jest.fn();
    const editShiftTemplateModal = mount(<EditShiftTemplateModal
      {...baseProps}
      shiftTemplate={baseShiftTemplate}
      isOpen
      onSave={jest.fn()}
      onClose={onClose}
      onDelete={jest.fn()}
    />);

    editShiftTemplateModal.find('button[aria-label="Close"]').simulate('click');
    expect(onClose).toBeCalled();
  });

  it('should call onDelete when the delete button is clicked', () => {
    const onDelete = jest.fn();
    const editShiftTemplateModal = mount(<EditShiftTemplateModal
      {...baseProps}
      shiftTemplate={baseShiftTemplate}
      isOpen
      onSave={jest.fn()}
      onClose={jest.fn()}
      onDelete={onDelete}
    />);

    editShiftTemplateModal.find('button[aria-label="Delete"]').simulate('click');
    expect(onDelete).toBeCalled();
  });

  it('should prevent changing URL when completed', () => {
    const event = {
      preventDefault: jest.fn(),
    };
    const editShiftTemplateModal = shallow(<EditShiftTemplateModal
      {...baseProps}
      shiftTemplate={baseShiftTemplate}
      isOpen
      onSave={jest.fn()}
      onClose={jest.fn()}
      onDelete={jest.fn()}
    />);
    editShiftTemplateModal.find('Form').simulate('submit', event);
    expect(event.preventDefault).toBeCalled();
  });

  it('should update the start day offset when the start day offset is changed', () => {
    const editShiftTemplateModal = shallow(<EditShiftTemplateModal
      {...baseProps}
      isOpen
      onSave={jest.fn()}
      onClose={jest.fn()}
      onDelete={jest.fn()}
    />);
    editShiftTemplateModal.find('TextInput[aria-label="Start Day Offset"]').simulate('change', 2);
    expect(editShiftTemplateModal.state('editedValue')).toEqual({
      startDayOffset: 1,
    });
    editShiftTemplateModal.find('TextInput[aria-label="Start Day Offset"]').simulate('change', undefined);
    expect(editShiftTemplateModal.state('editedValue')).toEqual({
      startDayOffset: undefined,
    });
  });

  it('should update the start time when the start time is changed', () => {
    const editShiftTemplateModal = shallow(<EditShiftTemplateModal
      {...baseProps}
      isOpen
      onSave={jest.fn()}
      onClose={jest.fn()}
      onDelete={jest.fn()}
    />);
    editShiftTemplateModal.find('TextInput[aria-label="Start Time"]').simulate('change', '13:30');
    expect(editShiftTemplateModal.state('editedValue')).toEqual({
      startTime: { hours: 13, minutes: 30 },
    });
  });

  it('should update the end day offset when the end day offset is changed', () => {
    const editShiftTemplateModal = shallow(<EditShiftTemplateModal
      {...baseProps}
      isOpen
      onSave={jest.fn()}
      onClose={jest.fn()}
      onDelete={jest.fn()}
    />);
    editShiftTemplateModal.find('TextInput[aria-label="End Day Offset"]').simulate('change', 4);
    expect(editShiftTemplateModal.state('editedValue')).toEqual({
      endDayOffset: 3,
    });
    editShiftTemplateModal.find('TextInput[aria-label="End Day Offset"]').simulate('change', undefined);
    expect(editShiftTemplateModal.state('editedValue')).toEqual({
      endDayOffset: undefined,
    });
  });

  it('should update the end time when the end time is changed', () => {
    const editShiftTemplateModal = shallow(<EditShiftTemplateModal
      {...baseProps}
      isOpen
      onSave={jest.fn()}
      onClose={jest.fn()}
      onDelete={jest.fn()}
    />);
    editShiftTemplateModal.find('TextInput[aria-label="End Time"]').simulate('change', '13:30');
    expect(editShiftTemplateModal.state('editedValue')).toEqual({
      endTime: { hours: 13, minutes: 30 },
    });
  });

  it('should update the spot when the spot is changed', () => {
    const editShiftTemplateModal = shallow(<EditShiftTemplateModal
      {...baseProps}
      isOpen
      onSave={jest.fn()}
      onClose={jest.fn()}
      onDelete={jest.fn()}
    />);
    editShiftTemplateModal.find('TypeaheadSelectInput[aria-label="Spot"]').simulate('change', spot);
    expect(editShiftTemplateModal.state('editedValue')).toEqual({
      spot,
    });
  });

  it('should update the employee when the employee is changed', () => {
    const editShiftTemplateModal = shallow(<EditShiftTemplateModal
      {...baseProps}
      isOpen
      onSave={jest.fn()}
      onClose={jest.fn()}
      onDelete={jest.fn()}
    />);
    editShiftTemplateModal.find('TypeaheadSelectInput[aria-label="Employee"]').simulate('change', employee);
    expect(editShiftTemplateModal.state('editedValue')).toEqual({
      rotationEmployee: employee,
    });

    editShiftTemplateModal.find('TypeaheadSelectInput[aria-label="Employee"]').simulate('change', undefined);
    expect(editShiftTemplateModal.state('editedValue')).toEqual({
      rotationEmployee: null,
    });
  });

  it('shiftTemplateDataToShiftTemplate should work as expected', () => {
    expect(shiftTemplateDataToShiftTemplate({
      tenantId: 0,
      id: 1,
      version: 2,
      spot,
      rotationEmployee: employee,
      startDayOffset: 2,
      startTime: { hours: 9, minutes: 0 },
      endDayOffset: 3,
      endTime: { hours: 9, minutes: 0 },
    }, rotationLength)).toEqual({
      tenantId: 0,
      id: 1,
      version: 2,
      spot,
      rotationEmployee: employee,
      durationBetweenRotationStartAndTemplateStart: moment.duration(2, 'd').add(9, 'h'),
      shiftTemplateDuration: moment.duration(1, 'd'),
    });

    expect(shiftTemplateDataToShiftTemplate({
      tenantId: 0,
      id: 1,
      version: 2,
      spot,
      rotationEmployee: null,
      startDayOffset: rotationLength - 1,
      startTime: { hours: 9, minutes: 0 },
      endDayOffset: 0,
      endTime: { hours: 9, minutes: 0 },
    }, rotationLength)).toEqual({
      tenantId: 0,
      id: 1,
      version: 2,
      spot,
      rotationEmployee: null,
      durationBetweenRotationStartAndTemplateStart: moment.duration(rotationLength - 1, 'd').add(9, 'h'),
      shiftTemplateDuration: moment.duration(1, 'd'),
    });
  });

  it('shiftTemplateToShiftTemplateData should work as expected', () => {
    expect(shiftTemplateToShiftTemplateData(baseShiftTemplate, rotationLength)).toEqual({
      tenantId: 0,
      id: 1,
      version: 0,
      spot,
      rotationEmployee: employee,
      startDayOffset: 1,
      startTime: { hours: 9, minutes: 0 },
      endDayOffset: 1,
      endTime: { hours: 17, minutes: 0 },
    });

    expect(shiftTemplateToShiftTemplateData({
      tenantId: 0,
      id: 1,
      version: 2,
      spot,
      rotationEmployee: null,
      durationBetweenRotationStartAndTemplateStart: moment.duration(rotationLength - 1, 'days').add(9, 'hours'),
      shiftTemplateDuration: moment.duration(24, 'hours'),
    }, rotationLength)).toEqual({
      tenantId: 0,
      id: 1,
      version: 2,
      spot,
      rotationEmployee: null,
      startDayOffset: rotationLength - 1,
      startTime: { hours: 9, minutes: 0 },
      endDayOffset: 0,
      endTime: { hours: 9, minutes: 0 },
    });
  });
});

const spot: Spot = {
  tenantId: 0,
  id: 2,
  version: 0,
  name: 'Spot',
  requiredSkillSet: [
    {
      tenantId: 0,
      id: 3,
      version: 0,
      name: 'Skill',
    },
  ],
};

const employee: Employee = {
  tenantId: 0,
  id: 4,
  version: 0,
  name: 'Employee 1',
  contract: {
    tenantId: 0,
    id: 5,
    version: 0,
    name: 'Basic Contract',
    maximumMinutesPerDay: 10,
    maximumMinutesPerWeek: 70,
    maximumMinutesPerMonth: 500,
    maximumMinutesPerYear: 6000,
  },
  skillProficiencySet: [{
    tenantId: 0,
    id: 6,
    version: 0,
    name: 'Not Required Skill',
  }],
};

const baseShiftTemplate: ShiftTemplate = {
  tenantId: 0,
  id: 1,
  version: 0,
  durationBetweenRotationStartAndTemplateStart: moment.duration(1, 'd').add(9, 'h'),
  shiftTemplateDuration: moment.duration(8, 'h'),
  spot,
  rotationEmployee: employee,
};

const rotationLength = 24;

const baseProps: Props & WithTranslation = {
  ...useTranslation(),
  tReady: true,
  tenantId: 0,
  isOpen: true,
  spotList: [spot],
  employeeList: [employee],
  rotationLength,
  onSave: jest.fn(),
  onDelete: jest.fn(),
  onClose: jest.fn(),
};
