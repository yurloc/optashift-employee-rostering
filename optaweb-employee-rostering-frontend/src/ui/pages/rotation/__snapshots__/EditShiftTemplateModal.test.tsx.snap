// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Shift Template Modal should render correctly when closed 1`] = `
<Modal
  actions={
    Array [
      <Button
        aria-label="Close Modal"
        onClick={[MockFunction]}
        variant="tertiary"
      >
        Trans(i18nKey=close)
      </Button>,
      <Button
        aria-label="Save"
        onClick={[Function]}
      >
        Trans(i18nKey=save)
      </Button>,
    ]
  }
  ariaDescribedById=""
  className=""
  hideTitle={false}
  isLarge={false}
  isOpen={false}
  isSmall={true}
  onClose={[MockFunction]}
  showClose={true}
  title="Trans(i18nKey=createShiftTemplate)"
>
  <Form
    id="modal-element"
    key="0"
    onSubmit={[Function]}
  >
    <InputGroup>
      <Label>
        Trans(i18nKey=spot)
      </Label>
      <TypeaheadSelectInput
        aria-label="Spot"
        emptyText="Trans(i18nKey=selectSpot)"
        onChange={[Function]}
        optionToStringMap={[Function]}
        options={
          Array [
            Object {
              "id": 2,
              "name": "Spot",
              "requiredSkillSet": Array [
                Object {
                  "id": 3,
                  "name": "Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
          ]
        }
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Trans(i18nKey=startDayOffset)
      </Label>
      <TextInput
        aria-label="Start Day Offset"
        className=""
        isDisabled={false}
        isReadOnly={false}
        isRequired={false}
        isValid={true}
        max={24}
        min={1}
        onChange={[Function]}
        type="number"
      />
      <Label>
        Trans(i18nKey=startTime)
      </Label>
      <TextInput
        aria-label="Start Time"
        className=""
        isDisabled={false}
        isReadOnly={false}
        isRequired={false}
        isValid={true}
        onChange={[Function]}
        type="time"
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Trans(i18nKey=endDayOffset)
      </Label>
      <TextInput
        aria-label="End Day Offset"
        className=""
        isDisabled={false}
        isReadOnly={false}
        isRequired={false}
        isValid={true}
        max={24}
        min={1}
        onChange={[Function]}
        type="number"
      />
      <Label>
        Trans(i18nKey=endTime)
      </Label>
      <TextInput
        aria-label="End Time"
        className=""
        isDisabled={false}
        isReadOnly={false}
        isRequired={false}
        isValid={true}
        onChange={[Function]}
        type="time"
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Trans(i18nKey=rotationEmployee)
      </Label>
      <TypeaheadSelectInput
        aria-label="Employee"
        emptyText="Trans(i18nKey=unassigned)"
        onChange={[Function]}
        optionToStringMap={[Function]}
        optional={true}
        options={
          Array [
            undefined,
            Object {
              "contract": Object {
                "id": 5,
                "maximumMinutesPerDay": 10,
                "maximumMinutesPerMonth": 500,
                "maximumMinutesPerWeek": 70,
                "maximumMinutesPerYear": 6000,
                "name": "Basic Contract",
                "tenantId": 0,
                "version": 0,
              },
              "id": 4,
              "name": "Employee 1",
              "skillProficiencySet": Array [
                Object {
                  "id": 6,
                  "name": "Not Required Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
          ]
        }
      />
    </InputGroup>
  </Form>
</Modal>
`;

exports[`Edit Shift Template Modal should render correctly when opened 1`] = `
<Modal
  actions={
    Array [
      <Button
        aria-label="Close Modal"
        onClick={[MockFunction]}
        variant="tertiary"
      >
        Trans(i18nKey=close)
      </Button>,
      <Button
        aria-label="Save"
        onClick={[Function]}
      >
        Trans(i18nKey=save)
      </Button>,
    ]
  }
  ariaDescribedById=""
  className=""
  hideTitle={false}
  isLarge={false}
  isOpen={true}
  isSmall={true}
  onClose={[MockFunction]}
  showClose={true}
  title="Trans(i18nKey=createShiftTemplate)"
>
  <Form
    id="modal-element"
    key="0"
    onSubmit={[Function]}
  >
    <InputGroup>
      <Label>
        Trans(i18nKey=spot)
      </Label>
      <TypeaheadSelectInput
        aria-label="Spot"
        emptyText="Trans(i18nKey=selectSpot)"
        onChange={[Function]}
        optionToStringMap={[Function]}
        options={
          Array [
            Object {
              "id": 2,
              "name": "Spot",
              "requiredSkillSet": Array [
                Object {
                  "id": 3,
                  "name": "Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
          ]
        }
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Trans(i18nKey=startDayOffset)
      </Label>
      <TextInput
        aria-label="Start Day Offset"
        className=""
        isDisabled={false}
        isReadOnly={false}
        isRequired={false}
        isValid={true}
        max={24}
        min={1}
        onChange={[Function]}
        type="number"
      />
      <Label>
        Trans(i18nKey=startTime)
      </Label>
      <TextInput
        aria-label="Start Time"
        className=""
        isDisabled={false}
        isReadOnly={false}
        isRequired={false}
        isValid={true}
        onChange={[Function]}
        type="time"
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Trans(i18nKey=endDayOffset)
      </Label>
      <TextInput
        aria-label="End Day Offset"
        className=""
        isDisabled={false}
        isReadOnly={false}
        isRequired={false}
        isValid={true}
        max={24}
        min={1}
        onChange={[Function]}
        type="number"
      />
      <Label>
        Trans(i18nKey=endTime)
      </Label>
      <TextInput
        aria-label="End Time"
        className=""
        isDisabled={false}
        isReadOnly={false}
        isRequired={false}
        isValid={true}
        onChange={[Function]}
        type="time"
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Trans(i18nKey=rotationEmployee)
      </Label>
      <TypeaheadSelectInput
        aria-label="Employee"
        emptyText="Trans(i18nKey=unassigned)"
        onChange={[Function]}
        optionToStringMap={[Function]}
        optional={true}
        options={
          Array [
            undefined,
            Object {
              "contract": Object {
                "id": 5,
                "maximumMinutesPerDay": 10,
                "maximumMinutesPerMonth": 500,
                "maximumMinutesPerWeek": 70,
                "maximumMinutesPerYear": 6000,
                "name": "Basic Contract",
                "tenantId": 0,
                "version": 0,
              },
              "id": 4,
              "name": "Employee 1",
              "skillProficiencySet": Array [
                Object {
                  "id": 6,
                  "name": "Not Required Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
          ]
        }
      />
    </InputGroup>
  </Form>
</Modal>
`;

exports[`Edit Shift Template Modal should render correctly with a shift template when opened 1`] = `
<Modal
  actions={
    Array [
      <Button
        aria-label="Close Modal"
        onClick={[MockFunction]}
        variant="tertiary"
      >
        Trans(i18nKey=close)
      </Button>,
      <Button
        aria-label="Delete"
        onClick={[Function]}
        variant="danger"
      >
        Trans(i18nKey=delete)
      </Button>,
      <Button
        aria-label="Save"
        onClick={[Function]}
      >
        Trans(i18nKey=save)
      </Button>,
    ]
  }
  ariaDescribedById=""
  className=""
  hideTitle={false}
  isLarge={false}
  isOpen={true}
  isSmall={true}
  onClose={[MockFunction]}
  showClose={true}
  title="Trans(i18nKey=editShiftTemplate)"
>
  <Form
    id="modal-element"
    key="0"
    onSubmit={[Function]}
  >
    <InputGroup>
      <Label>
        Trans(i18nKey=spot)
      </Label>
      <TypeaheadSelectInput
        aria-label="Spot"
        emptyText="Trans(i18nKey=selectSpot)"
        onChange={[Function]}
        optionToStringMap={[Function]}
        options={
          Array [
            Object {
              "id": 2,
              "name": "Spot",
              "requiredSkillSet": Array [
                Object {
                  "id": 3,
                  "name": "Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
          ]
        }
        value={
          Object {
            "id": 2,
            "name": "Spot",
            "requiredSkillSet": Array [
              Object {
                "id": 3,
                "name": "Skill",
                "tenantId": 0,
                "version": 0,
              },
            ],
            "tenantId": 0,
            "version": 0,
          }
        }
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Trans(i18nKey=startDayOffset)
      </Label>
      <TextInput
        aria-label="Start Day Offset"
        className=""
        defaultValue="2"
        isDisabled={false}
        isReadOnly={false}
        isRequired={false}
        isValid={true}
        max={24}
        min={1}
        onChange={[Function]}
        type="number"
      />
      <Label>
        Trans(i18nKey=startTime)
      </Label>
      <TextInput
        aria-label="Start Time"
        className=""
        defaultValue="09:00"
        isDisabled={false}
        isReadOnly={false}
        isRequired={false}
        isValid={true}
        onChange={[Function]}
        type="time"
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Trans(i18nKey=endDayOffset)
      </Label>
      <TextInput
        aria-label="End Day Offset"
        className=""
        defaultValue="2"
        isDisabled={false}
        isReadOnly={false}
        isRequired={false}
        isValid={true}
        max={24}
        min={1}
        onChange={[Function]}
        type="number"
      />
      <Label>
        Trans(i18nKey=endTime)
      </Label>
      <TextInput
        aria-label="End Time"
        className=""
        defaultValue="17:00"
        isDisabled={false}
        isReadOnly={false}
        isRequired={false}
        isValid={true}
        onChange={[Function]}
        type="time"
      />
    </InputGroup>
    <InputGroup>
      <Label>
        Trans(i18nKey=rotationEmployee)
      </Label>
      <TypeaheadSelectInput
        aria-label="Employee"
        emptyText="Trans(i18nKey=unassigned)"
        onChange={[Function]}
        optionToStringMap={[Function]}
        optional={true}
        options={
          Array [
            undefined,
            Object {
              "contract": Object {
                "id": 5,
                "maximumMinutesPerDay": 10,
                "maximumMinutesPerMonth": 500,
                "maximumMinutesPerWeek": 70,
                "maximumMinutesPerYear": 6000,
                "name": "Basic Contract",
                "tenantId": 0,
                "version": 0,
              },
              "id": 4,
              "name": "Employee 1",
              "skillProficiencySet": Array [
                Object {
                  "id": 6,
                  "name": "Not Required Skill",
                  "tenantId": 0,
                  "version": 0,
                },
              ],
              "tenantId": 0,
              "version": 0,
            },
          ]
        }
        value={
          Object {
            "contract": Object {
              "id": 5,
              "maximumMinutesPerDay": 10,
              "maximumMinutesPerMonth": 500,
              "maximumMinutesPerWeek": 70,
              "maximumMinutesPerYear": 6000,
              "name": "Basic Contract",
              "tenantId": 0,
              "version": 0,
            },
            "id": 4,
            "name": "Employee 1",
            "skillProficiencySet": Array [
              Object {
                "id": 6,
                "name": "Not Required Skill",
                "tenantId": 0,
                "version": 0,
              },
            ],
            "tenantId": 0,
            "version": 0,
          }
        }
      />
    </InputGroup>
  </Form>
</Modal>
`;
